pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/v2/PersistentTopics.java

```diff
@@ -1318,9 +1318,9 @@ public void createSubscription(
             @ApiParam(value = "Is authentication required to perform this operation")
             @QueryParam("authoritative") @DefaultValue("false") boolean authoritative,
             @ApiParam(name = "messageId", value = "messageId where to create the subscription. "
-                    + "It can be 'latest', 'earliest' or (ledgerId:entryId)",
+                    + "It can be 'latest', 'earliest' or (ledgerId:entryId:partitionIndex)",
                     defaultValue = "latest",
-                    allowableValues = "latest, earliest, ledgerId:entryId"
+                    allowableValues = "latest, earliest, (ledgerId:entryId:partitionIndex)"
```

Comment:
We should use the `ResetCursorData` here, for couple reasons:

1. In pulsar, the default behavior is to create the subscription on the latest position, so if users did not provide the messageId, it means creating a subscription by using the latest position.
2. `isExcluded` and `batchIndex` should be also considered when creating a subscription.
3. Follow the same API def with PersistentTopics.resetCursorOnPosition(), for passing the position information, we should provide a consistent way.

Reply:
I agree. `ResetCursorData` is more suitable than `MessageIdImpl`.